src/Infrastructure/Repositories/LocalFilesRepository.php

```diff
@@ -40,33 +44,76 @@ public function getDefaultDirectory(): string
 
     public function getFiles(): array
     {
-        if (null === $this->files) {
-            $this->files = iterator_to_array($this->finder->getIterator(), true);
+        if ($this->files === null) {
+            $this->files = $this->getIterator();
         }
 
         return $this->files;
     }
 
-    public function within(string $path, array $exclude = []): FilesRepository
+    public function within(array $paths, array $exclude = []): FilesRepository
     {
-        if (! is_dir($path) && is_file($path)) {
+        foreach ($paths as $path) {
             $pathInfo = pathinfo($path);
-            $this->finder = Finder::create()
-                ->in($pathInfo['dirname'])
-                ->name($pathInfo['basename']);
 
-            return $this;
+            if (! is_dir($path) && is_file($path)) {
+                $this->fileList['dirname'][] = $pathInfo['dirname'];
+                $this->fileList['basename'][] = $pathInfo['basename'];
+            } else {
+                $this->directoryList[] = $pathInfo['dirname'] . DIRECTORY_SEPARATOR . $pathInfo['basename'];
+            }
         }
-        $this->finder->in([$path])->notPath($exclude);
+
+        $directoryFiles = $this->directoryList === null ? [] : $this->getDirectoryFiles($exclude);
+        $singleFiles = $this->fileList === null ? [] : $this->getSingleFiles();
+
+        $this->files = array_merge($directoryFiles, $singleFiles);
+
+        return $this;
+    }
+
+    /**
+     * @param array<string> $exclude
+     *
+     * @return array<\Symfony\Component\Finder\SplFileInfo>
+     */
+    private function getDirectoryFiles(array $exclude = []): array
+    {
+        $this->finder = Finder::create()
+            ->files()
+            ->name(['*.php'])
+            ->exclude(['vendor', 'tests', 'Tests', 'test', 'Test'])
+            ->notName(['*.blade.php'])
+            ->ignoreUnreadableDirs()
+            ->in($this->directoryList)
+            ->notPath($exclude);
 
         foreach ($exclude as $value) {
             if (substr($value, -4) === '.php') {
                 $this->finder->notName($value);
             }
         }
 
-        $this->files = iterator_to_array($this->finder->getIterator(), true);
+        return $this->getIterator();
+    }
 
-        return $this;
+    /**
+     * @return array<\Symfony\Component\Finder\SplFileInfo>
+     */
+    private function getSingleFiles(): array
+    {
+        $this->finder = Finder::create()
+            ->in($this->fileList['dirname'])
+            ->name($this->fileList['basename']);
+
+        return $this->getIterator();
+    }
+
+    /**
+     * @return array<\Symfony\Component\Finder\SplFileInfo>
+     */
+    private function getIterator(): array
```

Comment:
I would named this getFilesList instead getIterator ?

Reply:
Done